utils/s3/client.cc

```diff
@@ -223,6 +234,92 @@ future<client::stats> client::get_object_stats(sstring object_name) {
     co_return st;
 }
 
+static rapidxml::xml_node<>* first_node_of(rapidxml::xml_node<>* root,
+                                           std::initializer_list<std::string_view> names) {
+    assert(root);
+    auto* node = root;
+    for (auto name : names) {
+        node = node->first_node(name.data(), name.size());
+        if (!node) {
+            throw std::runtime_error(fmt::format("'{}' is not found", name));
+        }
+    }
+    return node;
+}
+
+static tag_set parse_tagging(sstring& body) {
+    auto doc = std::make_unique<rapidxml::xml_document<>>();
+    try {
+        doc->parse<0>(body.data());
+    } catch (const rapidxml::parse_error& e) {
+        s3l.warn("cannnot parse tagging response: {}", e.what());
+        throw std::runtime_error("cannot parse tagging response");
+    }
+    tag_set tags;
+    auto tagset_node = first_node_of(doc.get(), {"Tagging", "TagSet"});
+    for (auto tag_node = tagset_node->first_node("Tag"); tag_node; tag_node = tag_node->next_sibling()) {
+        auto key = tag_node->first_node("Key")->value();
+        auto value = tag_node->first_node("Value")->value();
+        tags.emplace_back(tag{key, value});
+    }
+    return tags;
+}
+
+future<tag_set> client::get_object_tagging(sstring object_name) {
+    // see https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectTagging.html
+    auto req = http::request::make("GET", _host, object_name);
+    req.query_parameters["tagging"] = "";
+    s3l.trace("GET {} tagging", object_name);
+    tag_set tags;
+    co_await make_request(std::move(req),
+                          [&tags] (const http::reply& reply, input_stream<char>&& in) mutable -> future<> {
+        auto& retval = tags;
+        auto input = std::move(in);
+        auto body = co_await util::read_entire_stream_contiguous(input);
+        retval = parse_tagging(body);
+    }, http::reply::status_type::ok);
```

Comment:
nit: `status_type::ok` is default arg. Can be a followup pr

Reply:
will drop it tomorrow.